### DISCUSSION TITLE
should I just wait or do something to start the job?

### DISCUSSION THREAD

**Discussion by kexinliu0001** [nextpyp/.github#27] [Q&A]
<img width="854" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/eeb2fb44-ee64-4bf2-a6ef-2246ff903a7f">
I'm importing the raw data, but the job has been in  Scheduled state for a long time (already half an hour). 
(I used to import the raw data in a previous project, and it is very fast.)
should I just wait or do something to start this job?

**Comment by kexinliu0001** [comment]
I restarted nextpyp (systemctl restart nextPYP) and solved the problem.

is there any other solution?

**Reply by tofushan** [reply]
Are you using standalone mode or SLURM? If it is latter, waiting for the resources is normal if currently no enough SLURM resources available to you. To solve this, you may need to consult your admin to increase your quota etc.

**Reply by kexinliu0001** [reply]
I'm using standalone mode, and the state of the jobs is always sleep before I restart nextpyp and resubmit the job.
<img width="1317" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/8f7d163d-0553-40ce-aa69-9a497c60907e">
<img width="1300" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/e97f0a1d-64e2-456e-82b9-ac48b64bae54">

**Reply by cuchaz** [reply]
Restarting nextpyp should work around the issue.

That waiting message means the website has submitted the job to the job runner and is just waiting for the job runner (the `hostprocessor` script in standalone mode) to actually start the job process. If `hostprocessor` didn't actually start the job process for some reason, the website will have no way of knowing what happened, so it will just wait forever.

If something went wrong with your `hostprocessor` script, there may be more information in the hostprocessor [log file](https://nextpyp.app/files/pyp/latest/docs/install/troubleshooting.html).

**Reply by kexinliu0001** [reply]
I'm using the Standalone mode.
why is there still some logs with slurm in the log file:
<img width="1190" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/eb347012-bcd8-4c0c-bcad-6e94618186fe">

**Reply by cuchaz** [reply]
The hostprocessor tries to be compatible with the SLURM interface, so it uses the same environment variables to communicate with the guest process.

I don't see any errors in the screenshot you posted. Are there any errors anywhere else in the file? Posting the whole file (rather than a screenshot) would be more helpful.

**Reply by kexinliu0001** [reply]
[mongod.log](https://github.com/nextpyp/.github/files/13889976/mongod.log)
[hostprocessor.txt](https://github.com/nextpyp/.github/files/13890007/hostprocessor.txt)
[micromon.txt](https://github.com/nextpyp/.github/files/13890024/micromon.txt)

hi, thank you for your reply!
I'm not sure if the other log files are useful.

**Reply by cuchaz** [reply]
Thanks for sharing the logs. It looks like the website is starting up correctly and there are no errors reported in the hostprocessor log. But that probably just means the errors aren't captured in the log somehow.

There's a chance the problem may be related to the default shell in your OS. What operating system are you running nextPYP on? And what's the default shell there? This command should tell you what is the default shell: `ls -al /bin/sh`.

In the meantime, we can try to improve the error capture so the logs are more helpful.

I don't know if this is related or not, but some of the folders you're using contain complicated characters in the names. For example, `EMPIAR-10164_（HIV-1_Gag）` contains a [complicated parenthesis character](https://util.unicode.org/UnicodeJsps/character.jsp?a=FF08). Much of our code should be able to handle the complicated characters, but maybe some of it can't. As a workaround, you could try using just ASCII characters in the names and see if the problem goes away.

**Reply by kexinliu0001** [reply]
hi,

the default shell is :
lrwxrwxrwx. 1 root root 4 Apr  7  2022 /bin/sh -> bash

I created another project named test_HIV, and I'm importing the raw data.
but it keeps waiting:

<img width="1437" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/73b325e8-3d60-4195-9d8a-f588700291be">

this step should have been very fast based on the past experience...

**Reply by cuchaz** [reply]
Ok, thanks. That rules out some possibilities, but I'm still not sure what the problem is. I'll need to update our code to collect more information. I think some error messages are getting lost.

I'll be in touch later hopefully with a new version you can try out that will try to log the errors better.

**Reply by cuchaz** [reply]
Hello again. It's possible this problem could have been caused by bugs in our job cancel system. For example, canceling some other job may have put the website in an inconsistent state and caused this job to get stuck.

The next release will have some fixes for canceling jobs and that may avoid this problem for you. When the next release is out (hopefully soon?), give it a try and see if it solves the problem for you.

**Reply by abartesaghi** [reply]
The new release [0.6.0](https://nextpyp.app/files/pyp/0.6.0/docs/index.html) should address some of these issues. Let us know if you are still having problems after upgrading.
